Book/php7/internal_types/functions/callables.rst

```diff
@@ -0,0 +1,129 @@
+  PHP didn't have objects and class methods. It must be a string zval or an array following the same rules as
+  callables in PHP, namely the first index is a class or instance object, and the second one is the method name.
+  It can also be undefined if, and only if, a non empty FCC is provided.
+``retval``:
+  Mandatory field, which will contain the result of the PHP function
+``param_count``:
+  Mandatory field, the number of arguments that will be provided to this call to the function
+``params``:
+  contains positional arguments that will be provided to this call to the function,
+  can be ``NULL`` if ``param_count = 0``.
+``object``:
+  The object on which to call the method name stored in ``function_name``.
+``named_params``:
+  A HashTable containing named (or positional) arguments, added with PHP 8.0.
+
+Structure of ``zend_fcall_info_cache``
+--------------------------------------
+
+A ``zend_fcall_info_cache`` has the following structure::
+
+    typedef struct _zend_fcall_info_cache {
+        zend_function *function_handler;
+        zend_class_entry *calling_scope;
+        zend_class_entry *called_scope;
+        zend_object *object;
+    } zend_fcall_info_cache;
+
+Let detail the various FCC fields:
+
+``function_handler``:
+  The actual body of a PHP function that will be used by the VM, can be retrieved from the global function table
+  or a class function table (``zend_class_entry->function_table``).
+``object``:
+  If the function is an object method, this field is the relevant object.
+``called_scope``:
+  The scope in which to call the method, generally it's ``object->ce``.
+``calling_scope``:
+  The scope in which this call is made, only used by the VM.
+
+Note: to release a FCC you should use the ``void zend_release_fcall_info_cache(zend_fcall_info_cache *fcc)``
+function.
+
+.. warning:: Prior to PHP 7.3 there existed an ``initialized`` field. Now an FCC is considered initialized when
+  ``function_handler`` is set to a non-null pointer.
+
+Zend Engine API for callables
+-----------------------------
+
+The API can be found in the ``Zend_API.h`` header file.
+
+If you have a FCI/FCC pair for a callable you can call it directly by using the
+``zend_call_function(zend_fcall_info *fci, zend_fcall_info_cache *fci_cache)`` function.
+If you just need to change, or provide the arguments and return value you can use the
+``zend_fcall_info_call(zend_fcall_info *fci, zend_fcall_info_cache *fcc, zval *retval, zval *args)`` function.
+
+In the more likely case where you just have a callable zval, you have the choice of a couple different options
+depending on the use case.
+
+For a one-off call the ``call_user_function(function_table, object, function_name, retval_ptr, param_count, params)``
+and ``call_user_function_named(function_table, object, function_name, retval_ptr, param_count, params, named_params)``
+macro-functions will do the trick. Note that the ``function_table`` argument is not used and should always be ``NULL``.
```

Comment:
I think `function_table` was used prior to PHP 8.0.0.

Reply:
Only in PHP 7.0, 7.1 dropped the arg :/